include/swift/AST/Decl.h

```diff
@@ -4217,6 +4225,13 @@ class EnumDecl final : public NominalTypeDecl {
     return std::distance(eltRange.begin(), eltRange.end());
   }
 
+  /// Returns a range that iterates over all the elements of an enum that are
+  /// available during lowering.
```

Comment:
Can you add a comment here explaining that after lowering due to availability, the elements may no longer be available. This is a "what" comment, rather than a "why" comment (which is really what is needed here for future maintainers). Also, maybe since this is so special add something about availability into the API name? Your choice on that last one but I think it would add clarity.

Reply:
I'll update this comment to elaborate that this range includes the elements for which `isAvailableDuringLowering()` is true. I'll enhance the documentation for that method to better explain why you might care, but I'd rather not duplicate that in several places since it could easily get out of sync.